File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/dao/NovelDao.java
package br.com.algorit.mangaalert.roomdatabase.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import br.com.algorit.mangaalert.roomdatabase.model.Novel;

@Dao
public interface NovelDao {

    @Query("SELECT * FROM NOVEL ORDER BY NOME ASC")
    LiveData<List<Novel>> getAll();

    @Query("SELECT NOME FROM NOVEL WHERE NOME = :nome")
    String findByNome(String nome);

    @Insert
    void insert(Novel novel);

    @Query("UPDATE NOVEL SET CAPITULO = :capitulo WHERE NOME = :nome")
    void updateCapitulo(double capitulo, String nome);

    @Query("UPDATE NOVEL SET CHECKED = :checked WHERE NOME = :nome")
    void updateChecked(boolean checked, String nome);

    @Query("UPDATE NOVEL SET CHECKED = 0 WHERE CHECKED = 1")
    void uncheckAll();
}

File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/dao/NovelDao.kt
package br.com.algorit.mangaalert.roomdatabase.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import br.com.algorit.mangaalert.roomdatabase.model.Novel

@Dao
interface NovelDao {
    @get:Query("SELECT * FROM NOVEL ORDER BY NOME ASC")
    val all: LiveData<List<Novel>>

    @Query("SELECT NOME FROM NOVEL WHERE NOME = :nome")
    fun findByNome(nome: String): String

    @Insert
    fun insert(novel: Novel)

    @Query("UPDATE NOVEL SET CAPITULO = :capitulo WHERE NOME = :nome")
    fun updateCapitulo(capitulo: Double, nome: String)

    @Query("UPDATE NOVEL SET CHECKED = :checked WHERE NOME = :nome")
    fun updateChecked(checked: Boolean, nome: String)

    @Query("UPDATE NOVEL SET CHECKED = 0 WHERE CHECKED = 1")
    fun uncheckAll()
}